examples/key-value-store/src/net_ffi.rs

```diff
@@ -0,0 +1,64 @@
+use cxx::UniquePtr;
+use std::future::Future;
+
+#[cxx::bridge(namespace = "net_ffi")]
+mod ffi {
+    unsafe extern "C++" {
+        include!("key-value-store/src/net_ffi.hh");
+
+        type VoidFuture = crate::VoidFuture;
+        type StringFuture = crate::StringFuture;
+
+        #[rust_name = "ServerSocket"]
+        type server_socket;
+
+        #[rust_name = "ConnectedSocket"]
+        type connected_socket;
+
+        #[rust_name = "InputStream"]
+        type input_stream;
+
+        #[rust_name = "OutputStream"]
+        type output_stream;
+
+        fn listen(port: u16) -> UniquePtr<ServerSocket>;
+
+        fn accept(
+            server_socket: &UniquePtr<ServerSocket>,
+            socket: &mut UniquePtr<ConnectedSocket>,
+        ) -> VoidFuture;
+
+        fn get_input_stream(socket: &UniquePtr<ConnectedSocket>) -> UniquePtr<InputStream>;
+
+        fn get_output_stream(socket: &UniquePtr<ConnectedSocket>) -> UniquePtr<OutputStream>;
+
+        fn close_output_stream(socket: &UniquePtr<OutputStream>) -> VoidFuture;
+
+        fn read(socket: &UniquePtr<InputStream>) -> StringFuture;
+
+        fn write(socket: &UniquePtr<OutputStream>, msg: &str) -> VoidFuture;
+    }
+}
+
+#[cxx_async::bridge(namespace = net_ffi)]
+unsafe impl Future for VoidFuture {
+    type Output = ();
+}
+
+#[cxx_async::bridge(namespace = net_ffi)]
+unsafe impl Future for StringFuture {
+    type Output = String;
+}
+
+unsafe impl Send for ConnectedSocket {}
```

Comment:
This requires a comment explaining why it's sound to implement Send here.

Reply:
I got rid of `unsafe impl Send for ConnectedSocket {}`. `Disributed::map_single` required this. However, there is no reason to require the `Send` trait when `map_single` is called on the current shard. That is why the function `Disributed::map_current` has been created in #46, and I used it here.